### PR TITLE
get all required data for karmor probe when kubearmor is running

### PR DESCRIPTION
**Pull request by Essietom**
Signed-off-by: essietom <essietom@gmail.com>

### CODE REVIEW

File: KubeArmor/core/kubeArmor.go [hunk 1]
```diff
@@ -635,3 +640,37 @@ func KubeArmor() {
 	// destroy the daemon
 	dm.DestroyKubeArmorDaemon()
 }
+
+// StrToFile Function
+func KarmorDataToFile(val KarmorData, destFile string) {
```
Comment (nyrahul) [inline KubeArmor/core/kubeArmor.go]: this function should go in `karmorprobedata.go` file.

File: KubeArmor/core/kubeArmor.go [hunk 2]
```diff
@@ -635,3 +640,37 @@ func KubeArmor() {
...
+	j, err := json.Marshal(val)
+	if err != nil {
+		log.Println(err)
```
Comment (nyrahul) [inline KubeArmor/core/kubeArmor.go]: return in case of this failure.

File: KubeArmor/core/kubeArmor.go [hunk 3]
```diff
@@ -635,3 +640,37 @@ func KubeArmor() {
...
+	}
+	if _, err := os.Stat(destFile); err != nil {
+		newFile, err := os.Create(filepath.Clean(destFile))
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Failed to create a file (%s, %s)\n", destFile, err.Error())
```
Comment (nyrahul) [inline KubeArmor/core/kubeArmor.go]: Why are we using `fmt` in someplaces and `log` in other cases?

File: KubeArmor/core/kubeArmor.go [hunk 4]
```diff
@@ -635,3 +640,37 @@ func KubeArmor() {
...
+		}
+		if err := newFile.Close(); err != nil {
+			fmt.Fprintf(os.Stderr, "Failed to close the file (%s, %s)\n", destFile, err.Error())
+		}
+	}
+
+	// #nosec
+	file, err := os.OpenFile(destFile, os.O_RDWR, 0644)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to open a file (%s, %s)\n", destFile, err.Error())
```
Comment (nyrahul) [inline KubeArmor/core/kubeArmor.go]: return in case of failure.

File: KubeArmor/core/kubeArmor.go (hunk 1 above)
Comment (nyrahul) [inline KubeArmor/core/kubeArmor.go]: I think this whole function can be reduced to couple of lines. Something like this:

```go
j, err := json.Marshal(val)
// failure handling
err := ioutil.WriteFile(destFile, j, 0644)
// failure handlling
```

Why use `os.Stat()`, `os.Create()`, `os.Close()` etc?

**Review by nyrahul — CHANGES_REQUESTED** [review]

**Comment by nyrahul** [comment]
As discussed on call, please handle:
- [ ] squash the commits
- [ ] resolve conflicts
- [ ] license headers .. change to 2022
- [ ] documentation change in kubearmor ([getting started guide](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/deployment_guide.md)) to suggest using probe in case users want to understand compatibility.

File: KubeArmor/core/karmorprobedata.go [hunk 5]
```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2021 Authors of KubeArmor
+
+package core
+
+import (
+	"encoding/json"
+	"io/ioutil"
```
Comment (daemon1024) [inline KubeArmor/core/karmorprobedata.go]: Please refer https://go.dev/doc/go1.16#ioutil for switching ioutil to suggested APIs.

**Pull request closed without merging**